generator/src/lib.rs

```diff
@@ -85,13 +103,43 @@ pub fn derive_parser(input: TokenStream, include_grammar: bool) -> TokenStream {
         }
     }
 
-    let pairs = match parser::parse(Rule::grammar_rules, &data) {
+    let raw_data = data.clone();
+    let mut pairs = match parser::parse(Rule::grammar_rules, &raw_data) {
         Ok(pairs) => pairs,
         Err(error) => panic!("error parsing \n{}", error.renamed_rules(rename_meta_rule)),
     };
 
+    // parse `include!("file.pest")` to load partial and replace data
+    // TODO: Try to avoid parse twice as much as possible
+    let mut partial_pairs = pairs.clone().flatten().peekable();
+    while let Some(pair) = partial_pairs.next() {
+        if pair.as_rule() == Rule::include {
+            if let Some(filename) = partial_pairs.peek() {
+                let filepath = partial_path(path.as_ref(), filename.as_str());
+                let partial_data = match read_file(&filepath) {
+                    Ok(data) => data,
+                    Err(error) => panic!("error opening {:?}: {}", filepath, error),
+                };
+
+                let (start, end) = (pair.as_span().start(), pair.as_span().end());
+
+                data.replace_range(start..end, &partial_data);
+                has_use = true;
+            }
+        }
+    }
+
+    if has_use {
+        // Re-parse the data again, after replacing the `use` statement
+        pairs = match parser::parse(Rule::grammar_rules, &data) {
+            Ok(pairs) => pairs,
+            Err(error) => panic!("error parsing \n{}", error.renamed_rules(rename_meta_rule)),
+        };
+    }
```

Comment:
Doing it this way rather than in a loop means that only a single layer of `include!` will be resolved; a `include!` of a file with an `include!` won't work.

At a minimum we need a test for this case. Ideally we'd make it work, but so long as it results in an error rather than silently causing corruption it's not too bad.